File: java_projects/glue_code/commons-validator/src/main/java/org/apache/commons/validator/ISBNValidator.java
package org.apache.commons.validator;

import org.graalvm.polyglot.Value;

public class ISBNValidator {
  private static Value clz =
      ContextInitializer.getPythonClass("/ISBNValidator.py", "ISBNValidator");
  private Value obj;

  public ISBNValidator(Value obj) {
    this.obj = obj;
  }

  public Value getPythonObject() {
    return obj;
  }

  public boolean isValid(String isbn) {
    //
    // return org.apache.commons.validator.routines.ISBNValidator.getInstance0()
    // .isValidISBN10(isbn);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("isValid", isbn).as(boolean.class);
  }

  public ISBNValidator() {
    //
    // super();
    //

    this.obj = clz.invokeMember("__init__");
  }
}

File: data/recomposed_projects/deepseek-coder-33b-instruct/commons-validator/src/main/org/apache/commons/validator/ISBNValidator.py
from __future__ import annotations
import io
from src.main.org.apache.commons.validator.routines.CodeValidator import *
from src.main.org.apache.commons.validator.routines.checkdigit.CheckDigit import *
from src.main.org.apache.commons.validator.routines.checkdigit.CheckDigitException import *
from src.main.org.apache.commons.validator.routines.checkdigit.EAN13CheckDigit import *
from src.main.org.apache.commons.validator.routines.checkdigit.ISBN10CheckDigit import *


class ISBNValidator:

    def isValid(self, isbn: str) -> bool:

        # Create an instance of ISBNValidator
        validator = ISBNValidator.getInstance0()

        # Check if the ISBN is valid
        return validator.isValidISBN10(isbn)

    def __init__(self) -> None:

        self.codeValidator = CodeValidator()
        self.ean13CheckDigit = EAN13CheckDigit()
        self.isbn10CheckDigit = ISBN10CheckDigit()
